### DISCUSSION TITLE
Using source branch as config for PR?

### DISCUSSION THREAD

**Discussion by jbergstroem** [chdsbd/kodiak#772] [Ideas]
Opening a discussion since I'm not sure which is the expected behavior.

When for instance doing changes to github actions in a PR, the source branch defines the behavior through the PR lifecycle. While doing a few iterations of my `kodiak.toml` today, I couldn't let go of the thought that perhaps this is a behavior I want from kodiak as well.

What does others think of this?

**Comment by chdsbd** [comment]
I think using the source branch of the PR would be a nicer experience, since you could easily test some `.kodiak.toml` changes.

My only concern is that a malicious user could auto merge their pull request via [`approve.auto_approve_usernames`](https://kodiakhq.com/docs/config-reference#approveauto_approve_usernames).

**Reply by jbergstroem** [reply]
> My only concern is that a malicious user could auto merge their pull request via [`approve.auto_approve_usernames`](https://kodiakhq.com/docs/config-reference#approveauto_approve_usernames).

Does this not apply to apps like Kodiak as well? https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#configuring-required-approval-for-workflows-from-public-forks

**Reply by chdsbd** [reply]
Yeah, that doesn't apply to Kodiak. CircleCI has a similar issue where I have to manually push a fork's branch to my repository to get the status checks to run.

**Reply by jbergstroem** [reply]
Ok, too bad. Perhaps make it an option?

**Reply by jbergstroem** [reply]
Just noticed, codecov exposes a few settings to do manage this: https://docs.codecov.com/docs/codecov-yaml#restricting-changes
